here-naksha-lib-ext-manager/src/main/java/com/here/naksha/lib/extmanager/ExtensionCache.java

```diff
@@ -83,15 +85,38 @@ protected void buildExtensionCache(ExtensionConfig extensionConfig) {
       publishIntoCache(result, extensionConfig);
     });
 
-    // Removing existing extension which has been removed from the configuration
-    List<String> extIds =
-        extensionConfig.getExtensions().stream().map(Extension::getId).toList();
+    // Removing existing extension which has been removed from the configuration.
+    List<String> extIds = extensionConfig.getExtensions().stream()
+        .map(extension -> extension.getEnv() + ":" + extension.getId())
+        .toList();
 
     for (String key : loaderCache.keySet()) {
       if (!extIds.contains(key)) {
-        loaderCache.remove(key);
-        PluginCache.removeExtensionCache(key);
-        logger.info("Extension {} removed from cache.", key);
+        KVPair<Extension, ClassLoader> kvPair = loaderCache.get(key);
+        Object instanceObj = kvPair.getValue();
+        Extension extension = kvPair.getKey();
+        final String extensionIdWthEnv = extension.getEnv() + ":" + extension.getId();
+        if (!isNullOrEmpty(extension.getInitClassName())) {
```

Comment:
Removal of extension from cache happens at two places now:
1) When extension config is removed
2) When extension config is updated

We can create common method which takes care of extension removal from cache (including IExtensionInit.close()) and use the same method from both places.

Reply:
Done.